boot/zephyr/main.c

```diff
@@ -147,8 +141,15 @@ static void do_boot(struct boot_rsp *rsp)
 #endif
 
 #ifdef CONFIG_BOOT_INTR_VEC_RELOC
-    VTOR = vt;
+#if defined(CONFIG_SW_VECTOR_RELAY)
+    _vector_table_pointer = vt;
+#ifdef CONFIG_CPU_CORTEX_M_HAS_VTOR
+    SCB->VTOR = (uint32_t)__vector_relay_table;
 #endif
+#elif defined(CONFIG_CPU_CORTEX_M_HAS_VTOR)
```

Comment:
This is weird @e-rk looks both the ifdef and elif have the same check

Reply:
> This is weird @e-rk looks both the ifdef and elif have the same check

The `elif` refers to the `#if defined(CONFIG_SW_VECTOR_RELAY)` above, so not the same checks.

Reply:
yes, realized that, sorry for the noise.